packages/plugin-core/src/commands/SetupWorkspace.ts

```diff
@@ -14,39 +14,104 @@ import { BasicCommand } from "./base";
 import { Logger } from "../logger";
 import { resolveTilde } from "@dendronhq/common-server";
 
-type CommandOpts = {
+type CommandInput = {
   rootDirRaw: string;
+  workspaceInitializer?: WorkspaceInitializer;
+  workspaceType?: WorkspaceType;
+};
+
+type CommandOpts = CommandInput & {
   vault?: DVault;
   skipOpenWs?: boolean;
   /**
    * override prompts
    */
   skipConfirmation?: boolean;
-  workspaceInitializer?: WorkspaceInitializer;
-  workspaceType?: WorkspaceType;
-};
-
-type CommandInput = {
-  rootDirRaw: string;
-  workspaceInitializer?: WorkspaceInitializer;
 };
 
 type CommandOutput = DVault[];
 
 export { CommandOpts as SetupWorkspaceOpts };
 
+const CODE_WS_LABEL = "Code";
+const CODE_WS_DETAIL = "New workspace";
+
+enum EXISTING_ROOT_ACTIONS {
+  CONTINUE = "continue",
+  DELETE = "delete",
+  ABORT = "abort",
+}
+
 export class SetupWorkspaceCommand extends BasicCommand<
   CommandOpts,
   CommandOutput
 > {
   key = DENDRON_COMMANDS.INIT_WS.key;
 
   async gatherInputs(): Promise<CommandInput | undefined> {
-    const rootDirRaw = await VSCodeUtils.gatherFolderPath({
-      default: path.join(resolveTilde("~"), "Dendron"),
-    });
-    if (_.isUndefined(rootDirRaw)) {
-      return;
+    let workspaceType = WorkspaceType.CODE;
+    let rootDirRaw: string | undefined;
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+    if (
+      workspaceFolders !== undefined &&
+      workspaceFolders.length > 0 &&
+      (await WorkspaceUtils.getWorkspaceType({
+        workspaceFolders,
+      })) === WorkspaceType.NONE
+    ) {
+      // If there's a non-Dendron workspace already open, offer to convert that to a Dendron workspace first
+      const initNative = await VSCodeUtils.showQuickPick(
```

Comment:
for labels, can we make them more consistent and give more context? Something like the following:

- Code Workspace  - [no description]
    - A dedicated IDE workspace for just your notes

- Native Workspace - [no description]
    - A Dendron workspace that can be used alongside your existing projects

Reply:
We can't exactly do that because here may be more than 1 native workspace option if there are multiple workspace folders set up. I tried to change it as close as I could get it:

![](https://i.imgur.com/M0ACzpL.png)

Reply:
Hmm, can we swap out the label? instead of the name of the folder, can we use "Native Workspace"

For the detail, quote the folder name (or bold it if it supports markdown modifiers)

Reply:
No markdown support so I just quoted it:

![](https://i.imgur.com/uZngOKa.png)

Thanks for helping iterate on this!

Reply:
any time. good job!